docs/main_concepts.md

```diff
@@ -25,30 +25,6 @@ Behind the scenes, Streamlit re-executes the entire Python script from top to bo
 
 What this all means is that Streamlit allows you to work in a fast interactive loop: you type some code, save, type some more code, save, and on-and-on until you're happy with your app. The idea is to use Streamlit as a place where you can understand your code, debug it, perfect it, and finally share your results.
 
-## Caching
-
-Caching is super important when it comes to speed - especially when dealing with large data sets. It's always a good idea to load data, cache what's expensive, visualize the data, then spruce things up with interaction.
-
-Streamlit's ability to quickly update and re-execute the whole app is great when you're working with a trivial amount of data, but when you have long-running computations, this can get costly and time consuming. Instead of re-executing, you can safely reuse data with [`st.cache`](api.html#streamlit.cache). `st.cache` is a data store that lets Streamlit apps safely and effortlessly persist information.
-
-When you mark a function with Streamlit's cache annotation, it tells Streamlit that whenever the function is called that it needs to check three things:
```

Comment:
This is different from above. Above we have **four** things. I assume the four is the latest, so we need to update this here with 4.